torch/csrc/distributed/rpc/init.cpp

```diff
@@ -444,6 +445,48 @@ If the future completes with an error, an exception is thrown.
           &ProcessGroupAgent::sync,
           py::call_guard<py::gil_scoped_release>());
 
+  // Base class: torch.distributed.rpc.RpcBackendOptions.
+  py::class_<TensorPipeRpcBackendOptions>(
+      module, "TensorPipeRpcBackendOptions", rpcBackendOptions)
+      .def(py::init<>())
+      .def_readwrite(
+          "worker_name_to_id", &TensorPipeRpcBackendOptions::workerNameToId);
```

Comment:
Added #37784 to track. This is partly related to the elasticity project, as both might require some notion of a store to share the worker information. cc @agolynski

Reply:
Yes, currently we will need users to provide this name-id mapping through rpc backend options. Here you mean we automatically collect these information through c10::store, remove, add or update nodes?

Reply:
it doesn't have to be `c10d::store`, it's just one option we discussed before, where the `TensorPipeRpcBackendOptions ` can contain some `init_method` to initialize/find the store, and then every worker writes its own name->id into the store. This might require some changes to the existing `c10d::Store`. If TensorPipe has better options to support this feature, it's definitely welcome.

I am raising this concern because it seems asking applications to provide a name->id map might be a regression on API convenience.